log/field.go

```diff
@@ -0,0 +1,41 @@
+package log
+
+import "github.com/marcelocantos/frozen"
+
+// Fields is an interface that represents a collection of key values
+type Fields interface {
+	GetKeyValues() []frozen.KeyValue
+}
+
+// SingleField represents a single key value
+type SingleField struct {
+	key   string
+	value interface{}
+}
+
+// MultipleFields is a type for map of string to interface, meant for type checking
+// you can use FromMap to create it or directly use MultipleFields{}
+type MultipleFields map[string]interface{}
+
+// NewField creates a Fields interface representing a single key value
+func NewField(key string, value interface{}) Fields {
+	return &SingleField{key: key, value: value}
+}
+
+func (sf *SingleField) GetKeyValues() []frozen.KeyValue {
+	return []frozen.KeyValue{frozen.KV(sf.key, sf.value)}
+}
+
+func FromMap(m map[string]interface{}) Fields {
+	return MultipleFields(m)
+}
+
+func (mf MultipleFields) GetKeyValues() []frozen.KeyValue {
+	i := 0
```

Comment:
why the counter? append() will only reeallocate if it needs to, the make with the size will work fine

Reply:
i thought assigning is better than append? https://stackoverflow.com/questions/38654729/golang-slice-append-vs-assign-performance

Reply:
whoops nevermind, missed some bits there about the reallocation, will add the changes

Reply:
Didnt wee already have the conversation about premature optimisation? write the code in the more natural way first

Reply:
added the change